File: Controllers/RandomController.cs
using System;
using Microsoft.AspNetCore.Mvc;
using System.Collections.Generic;
using System.IO;
using System.Net.Http.Headers;

namespace RandomAPI.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class RandomController : ControllerBase
    {
        private List<string> Phrases = new List<string> { "Phrase 1", "Phrase 2", "Phrase 3" };
        private List<string> Images = new List<string> { "image1.png","image2.png" };

        [HttpGet("randomphrase")]
        public IActionResult GetRandomPhrase()
        {
            var random = new Random();
            int index = random.Next(Phrases.Count);
            return Ok(Phrases[index]);
        }

        [HttpGet("randomimage")]
        public IActionResult GetRandomImage()
        {
            var random = new Random();
            int index = random.Next(Images.Count);
            var imageName = Images[index];
            var imagePath = Path.Combine("Images", imageName);
            var image = System.IO.File.OpenRead(imagePath);
            return File(image, "image/jpeg", imageName); // Renvoie le fichier d'image avec le nom du fichier
        }


    }
}

File: src/api/randomController.ts
import express, { Request, Response, Router } from 'express';
import path from 'path';

const router: Router = express.Router();

const Phrases: string[] = ["Phrase 1", "Phrase 2", "Phrase 3"];
const Images: string[] = ["image1.png", "image2.png"];

router.get('/randomphrase', (req: Request, res: Response) => {
  const randomIndex: number = Math.floor(Math.random() * Phrases.length);
  const randomPhrase: string = Phrases[randomIndex];
  res.json(randomPhrase);
});

router.get('/randomimage', (req: Request, res: Response) => {
  const randomIndex: number = Math.floor(Math.random() * Images.length);
  const randomImageName: string = Images[randomIndex];
  res.json({ imageName: randomImageName });
});


export default router;
